Documentation/ceph-csi-drivers.md

```diff
@@ -124,5 +124,152 @@ when pod get spawned and destroyed at time of pod delete.
+sidecar:
+
+* Edit the `rook-ceph-operator-config` configmap and add the
+ following configurations
+
+```bash
+kubectl edit cm rook-ceph-operator-config -n rook-ceph
+```
+
+* Add the following properties if not present:
+
+```yaml
+data:
+  CSI_ENABLE_CSIADDONS: "true"
+```
+
+* After updating the configmap with those settings, a new sidecar
+ should now start automatically in the CSI provisioner and nodeplugin pod.
+
+> NOTE: Make sure the version of ceph-csi used is v3.5.0+
+
+### Deploying the controller
+
+Users can deploy the controller by running the following command:
+
+```bash
+kubectl create -f https://github.com/csi-addons/kubernetes-csi-addons/deploy/controller/setup-controller.yaml
+```
+
+This creates the required crds and configure permissions.
+
+```bash
+namespace/csi-addons-system created
+customresourcedefinition.apiextensions.k8s.io/csiaddonsnodes.csiaddons.openshift.io configured
+customresourcedefinition.apiextensions.k8s.io/networkfences.csiaddons.openshift.io configured
+customresourcedefinition.apiextensions.k8s.io/reclaimspacecronjobs.csiaddons.openshift.io configured
+customresourcedefinition.apiextensions.k8s.io/reclaimspacejobs.csiaddons.openshift.io configured
+serviceaccount/csi-addons-controller-manager created
+role.rbac.authorization.k8s.io/csi-addons-leader-election-role created
+clusterrole.rbac.authorization.k8s.io/csi-addons-manager-role created
+clusterrole.rbac.authorization.k8s.io/csi-addons-metrics-reader created
+clusterrole.rbac.authorization.k8s.io/csi-addons-proxy-role created
+rolebinding.rbac.authorization.k8s.io/csi-addons-leader-election-rolebinding created
+clusterrolebinding.rbac.authorization.k8s.io/csi-addons-manager-rolebinding created
+clusterrolebinding.rbac.authorization.k8s.io/csi-addons-proxy-rolebinding created
+configmap/csi-addons-manager-config created
+service/csi-addons-controller-manager-metrics-service created
+deployment.apps/csi-addons-controller-manager created
+```
+
+### Reclaim Space Operation
+
+If a user wants to initiate a Reclaim Space operation, he will create a ReclaimSpaceJob
+CR that includes the reclaim space as an operation.
+
+```yaml
+apiVersion: csiaddons.openshift.io/v1alpha1
+kind: ReclaimSpaceJob
+metadata:
+  name: sample-1
+spec:
+  target:
+    persistentVolumeClaim: pvc-1
+  backOffLimit: 10
+  retryDeadlineSeconds: 900
+```
+
+A user that created the ReclaimSpaceJob can follow the progress, and
+check the result of the operation:
+
+```bash
+kubectl apply -f reclaimspace.yaml
+job.csiaddons.openshift.io/sample-1 created
+kubectl get reclaimspacejobs
+NAME                       COMPLETIONS   DURATION   AGE   SPACE RECLAIMED
+sample-1               0/1           6s         6s    -
+[admin@ocs]$ kubectl get reclaimspacejobs
+NAME                       COMPLETIONS   DURATION   AGE   SPACE RECLAIMED
+sample-1               1/1           10s        25s   153mb
+```
+
+### ReclaimSpaceCronJob
```

Comment:
Is there documentation for the csi addons in that repo? If so, it would make more sense for just mentioning basic features of the csi addons here in the rook docs, and link to the csi addons repo for more details. This seems like a lot of detail for the rook docs.

Reply:
okay sure.